crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -841,6 +842,20 @@ fn string_type() -> impl NoirParser<UnresolvedType> {
         .map(UnresolvedType::String)
 }
 
+fn format_string_type(
+    type_parser: impl NoirParser<UnresolvedType>,
+) -> impl NoirParser<UnresolvedType> {
+    keyword(Keyword::FormatString)
+        .ignore_then(
+            type_expression()
+                .or_not()
+                .then_ignore(just(Token::Comma))
+                .then(choice((type_parser.clone(), tuple_type(type_parser))))
```

Comment:
```suggestion
                .then(type_parser)
```
A tuple type is already a type, no need to parse both